.coveragerc

```diff
@@ -0,0 +1,29 @@
+[run]
+# We run multiple processes in parallel, so we need to make sure that we don't
+# write to the same file at the same time.
+parallel = true
+
+branch = true
+dynamic_context = test_function
+source = warehouse
+omit =
+  # We don't want to get coverage information for our migrations.
+  warehouse/migrations/*
+
+  # We don't want to actually cover our __main__.py file because it is hard
+  # to cover and it really just acts as a tiny shim to a function.
+  warehouse/__main__.py
+
+  # Again, tiny shim code that we don't actually need to test and trying to
+  # do so would just get in the way.
+  warehouse/wsgi.py
+
+  # And again, tiny shim code.
+  warehouse/celery.py
+
+[report]
+exclude_also =
+  class \w+\(Interface\):
+  if (typing\.)?TYPE_CHECKING:
+
+skip_covered = true
```

Comment:
JFYI: I think these can go into the pre-existing `pyproject.toml`, under `[tool.coverage.run]`: https://coverage.readthedocs.io/en/7.5.4/config.html#toml-syntax

Reply:
I (wrongly) assumed that the `COVERAGE_PROCESS_START` value could not be something else that a `.coveragerc` value so I ported the configuration to this file.